scripts/testing/tolerance_test.py

```diff
@@ -1,78 +1,221 @@
-            testing_values = testing_data[field_name][data_name] # set testing data names from testing data
-            if len(good_values) != len(testing_values): # if the number of values/indices in each list from the testing file does not match good file, error out
-            # NOTE: I wanted to put this closer to the beginning so it could error out earlier but I wasn't able to figure it out (yet)
-                print("ERROR: testing file does not have the same values of files as the good file")  
-                exit()
-                    
-        # ALTERNATIVE SOLUTION    
-        #    for good_value, testing_value in zip(good_values, testing_values): # iterate over both at once
-                #print(good_value, "->", testing_value)
-        #        if good_value != testing_value:
-        #            print("ERROR: testing file does not have the same number of files as the good file")
-        #        if math.isclose(good_value, testing_value, abs_tol = args.tolerance) == False:
-        #            print(testing_value, "is not correct")
-            
-            for value in range(len(good_values)): # iterates through values within the good_values list
-                #print(good_values[i], "->", testing_values[i])  # one integer, index on both lists, i.e. testing_values[0] goes with good_values[0]
-                if math.isclose(good_values[value], testing_values[value], abs_tol = args.tolerance) == False:
-                    print(testing_values[value], "in", field_name, data_name, "is not correct") # print which value from what dictionary field and data name is incorrect from the testing data
-                    
-           
-           
\ No newline at end of file
+# Return list of items in l1 that are missing from l2
+def list_missing(l1, l2):
+    missing = [x for x in l1 if x not in l2]
+    return missing
+
+# Ensure list of field names are equal and error out with useful message
+def ensure_field_names(baseline_field_names, test_field_names):
+    error_found = False
+
+    if len(baseline_field_names) == 0:
+        print("ERROR: Baseline file had no field names")
+        error_found = True
+    if len(test_field_names) == 0:
+        print("ERROR: Test file had no field names")
+        error_found = True
+
+    if error_found:
+        sys.exit(1)
+
+    if len(baseline_field_names) > len(test_field_names):
+        print("ERROR: Test file is missing field names that are in the baseline file")
+        missing = list_missing(baseline_field_names, test_field_names)
+        print("       Missing field names: {0}".format(", ".join(missing)))
+        error_found = True
+    elif len(baseline_field_names) < len(test_field_names):
+        print("ERROR: Test file has extra field names not in baseline file")
+        missing = list_missing(test_field_names, baseline_field_names)
+        print("       Extra field names: {0}".format(", ".join(missing)))
+        error_found = True
```

Comment:
this was the construction I was referring to in the meeting, it looks like 
```python
test_names = ["displacement", "velocity"]
baseline_names = ["temperature", "voltage"]
```
might just fall through these conditionals, despite there being differences between the two lists. Another implementation might be something like

```python
if (test_names != baseline_names):
    print("ERROR: field names are different than baseline names")
    print(f"names present in test, but not baseline: {set(test_names).difference(set(baseline_names))}")
    print(f"names present in baseline, but not test: {set(baseline_names).difference(set(test_names))}")
```